### PR TITLE
Add null check to string_view

### PR DESCRIPTION
**Pull request by Fernthedev**
There's a chance for undefined behaviour in il2cpp string allocations when the given string is null. This attempts to make it more explicit what should occur.

### CODE REVIEW

File: shared/utils/il2cpp-utils.hpp
Comment (sc2ad) [inline shared/utils/il2cpp-utils.hpp]: Use `u""` here to ensure we don't call through to the string_view one and then back over to the utf16 one

File: src/utils/typedefs-wrapper.cpp
Comment (sc2ad) [inline src/utils/typedefs-wrapper.cpp]: No need to use `\0` here, just an empty string already has an implicit `\0` at the end of it, so just use: `""`

File: src/utils/typedefs-wrapper.cpp
Comment (sc2ad) [inline src/utils/typedefs-wrapper.cpp]: ditto

**Review by sc2ad — APPROVED** [review]
Approved after feedback is addressed

**Comment by RedBrumbler** [comment]
Before merging things, I think it's warranted to think about whether a null string view should return a non-null StringW, as in my view this is weird behaviour. I think a null string view should return a null StringW.

For a final decision I would like to see the use case that brought this issue to light, and decide from there

**Comment by RedBrumbler** [comment]
as it turns out, the std::string and std::string_view behaviour for a default constructed (null) string view is to produce a valid empty string

image

**Comment by sc2ad** [comment]
As shown by your first log line (specifically with `a.data()`): [we see a null string](https://godbolt.org/z/ffTKMY98x) so it's clear there is AN issue with the existing implementation. Converting a string view with a null pointer to a stringw is fundamentally odd, though as you mentioned in C++, an empty (but valid) `std::string` is made. So, I'm going to merge this impl and we can revise further in the future.

**Comment by RedBrumbler** [comment]
yes we do see a null string there, but my point was that the standard library turns that into a valid empty string, I also still think it's weird behaviour but if it's what the standard library does I think we should do the same (as is done by merging this ofc)

**Pull request merged**
